File: PROG/Week05/Store.java
/**
 * The store has and sells one product: Sticky tape.
 *
 * You can sell and restock a product, view stock and view cash.
 */
public class Store {
    public static void main(String[] args) {
        new Store().use();
    }

    private Product stickyTape = new Product("Sticky tape", 2.99, 200);
    private CashRegister cashRegister = new CashRegister();

    public Store() {
        // No need to do anything..
    }

    public void use() {
        while (true) {
            System.out.print("Choice (s/r/v/c/x): ");
            char c = In.nextChar();

            switch (c) {
                case '?':
                    help();
                    break;
                case 's':
                    sell();
                    break;
                case 'r':
                    restock();
                    break;
                case 'v':
                    viewStock();
                    break;
                case 'c':
                    viewCash();
                    break;
                case 'x':
                    System.exit(0);
                    break;

                default:
                    break;
            }
        }
    }

    private void sell() {
        System.out.print("Number: ");
        int n = In.nextInt();
        try {
            double earned = stickyTape.sell(n);
            cashRegister.add(earned);
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        }
    }

    private void restock() {
        System.out.print("Number: ");
        int n = In.nextInt();
        stickyTape.restock(n);
    }

    private void viewStock() {
        System.out.println(stickyTape);
    }

    private void viewCash() {
        System.out.println(cashRegister);
    }

    private void help() {
        System.out.println("Menu options");
        System.out.println("s = sell");
        System.out.println("r = restock");
        System.out.println("v = view stock");
        System.out.println("c = view cash");
        System.out.println("x = exit");
    }
}

File: PROG/Week05/Store.py
from CashRegister import CashRegister
from Product import Product


class Store:
    _sticky_tape: Product
    _cash_register: CashRegister

    def __init__(self) -> None:
        self._sticky_tape = Product("Sticky tape", 2.99, 200)
        self._cash_register = CashRegister()

    def sell(self):
        n = int(input("Number: "))
        try:
            earned = self._sticky_tape.sell(n)
            self._cash_register.add(earned)
        except ValueError as e:
            print(e)

    def restock(self):
        n = int(input("Number: "))
        self._sticky_tape.add(n)

    def view_stock(self):
        print(self._sticky_tape)

    def view_cash(self):
        print(self._cash_register)

    def use(self):
        while True:
            c = input("Choice (s/r/v/c/x): ").lower()

            match c:
                case "?":
                    self.print_help()
                case "s":
                    self.sell()
                case "r":
                    self.restock()
                case "v":
                    self.view_stock()
                case "c":
                    self.view_cash()
                case "x":
                    exit(0)

    @staticmethod
    def print_help(_self=None):  # can use both self.print_help() and Store.print_help()
        print("Menu options")
        print("s = sell")
        print("r = restock")
        print("v = view stock")
        print("c = view cash")
        print("x = exit")


if __name__ == "__main__":
    Store().use()
